kotlinx-coroutines-core/common/src/flow/SharedFlow.kt

```diff
@@ -292,11 +292,23 @@ public fun <T> MutableSharedFlow(
 // ------------------------------------ Implementation ------------------------------------
 
 internal class SharedFlowSlot : AbstractSharedFlowSlot<SharedFlowImpl<*>>() {
+    /**
+     * The first index in the buffer that was not yet consumed by this collector,
+     * or -1 if the slot is not allocated to a collector.
+     */
     @JvmField
-    var index = -1L // current "to-be-emitted" index, -1 means the slot is free now
+    var index = -1L
 
+    /**
+     * The current continuation of the collector.
+     *
+     * This field is used by the collector each time when it can not acquire the next value and suspends while
+     * waiting for the next one.
```

Comment:
Used how? Please formulate more precisely, otherwise it adds more mystery than removes. ("The collector's continuation is saved/written/preserved here" or something)

"cannot", "suspends waiting for the next value to arrive."